eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -188,7 +188,8 @@ object Commitments {
       }
     }
 
-    val htlcValueInFlight = outgoingHtlcs.map(_.add.amountMsat).sum
+    // We can't obtain value in flight by `outgoingHtlcs.map(_.add.amountMsat).sum` because `outgoingHtlcs` is a Set where HTLCs may contain identical amounts and Set does not allow duplicates
+    val htlcValueInFlight = outgoingHtlcs.foldLeft(0 msat)(_ + _.add.amountMsat)
```

Comment:
How about this? It is compact enough to be duplicated on line 233 and being in-lined has less risk of being accidentally removed.

```suggestion
    val htlcValueInFlight = outgoingHtlcs.foldLeft(0 msat)(_ + _.add.amountMsat) // can't use .map(...).sum on a set because duplicates would be removed
```